esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -456,13 +531,37 @@ void EsdlServiceImpl::configureTargets(IPropertyTree *cfg, const char *service)
 
         MapStringToMyClass<IEspPlugin> localCppPluginMap;
 
+        IEsdlDefService* svcDef = m_esdl->queryService(service);
+        if (!svcDef)
+            throw MakeStringException(-1, "ESDL binding - service '%s' not in definition!", service);
+        const char*      svcAuthDef  = svcDef->queryProp(ANNOTATION_AUTH_FEATURE);
+        const char*      svcAuthBnd  = definition_cfg->queryProp("@" ANNOTATION_AUTH_FEATURE);
+        m_methodAccessMaps.kill();
+
         Owned<IPropertyTreeIterator> iter = m_pServiceMethodTargets->getElements("Target");
         ForEach(*iter)
         {
             IPropertyTree &methodCfg = iter->query();
             const char *method = methodCfg.queryProp("@name");
             if (!method || !*method)
-                throw MakeStringException(-1, "ESDL binding - found target method entry without name!");
+                throw MakeStringException(-1, "ESDL binding - found %s target method entry without name!", service);
+            IEsdlDefMethod* mthDef = svcDef->queryMethodByName(method);
+            if (!mthDef)
+                throw MakeStringException(-1, "ESDL binding - found %s target method entry '%s' not in definition!", service, method);
+
+            Owned<MethodAccessMap> authMap(new MethodAccessMap());
+            EsdlAccessMapReporter  reporter(*authMap.get(), g_reporterHelper.getLink());
+            EsdlAccessMapGenerator generator(g_scopeMapper, g_levelMapper, reporter);
+            generator.setVariable(Var_Service, service);
+            generator.setVariable(Var_Method, method);
+            generator.insertScope(Scope_EsdlService, svcAuthDef);
+            generator.insertScope(Scope_EsdlMethod, mthDef->queryMetaData(ANNOTATION_AUTH_FEATURE));
+            generator.insertScope(Scope_BindingService, svcAuthBnd);
+            generator.insertScope(Scope_BindingMethod, methodCfg.queryProp("@" ANNOTATION_AUTH_FEATURE));
+            if (!m_defaultFeatureAuth.isEmpty())
+                generator.setDefaultSecurity(m_defaultFeatureAuth);
```

Comment:
What is the effective generator DefaultSecurity if m_defaultFeatureAuth isEmpty()?

Reply:
The default behavior of the base class, TAccessMapGenerator, is to require security but specify no default. The existing use cases of this class set a default. This use case will set the same default, unless the user explicitly changes it in the static service configuration. And NO_DEFAULT can be used.

It is an error to require security and not specify security. For hidl, errors are not fatal. For ESDL, errors are fatal. The ESDL reporter handles enforcement by throwing an exception when the error is reported.